### ISSUE TITLE
R integration

### ISSUE THREAD

**Issue by rstub** [eddelbuettel/rcppziggurat#8]
The `ZigguratR`generator allows to use the uniform RNG from R in conjunction with the much faster Ziggurat method. Now it would be great if one could use the Ziggurat method directly via `rnorm()` by specifying `RNGkind(normal.kind = "user")`. The code to support this seems to be already present:

https://github.com/eddelbuettel/rcppziggurat/blob/8a3cade324c4de37b8b35e709a223cce925cb9b9/src/ziggurat.cpp#L248

However, it does not work on my system:
``` r
library(RcppZiggurat)
is.loaded("user_norm_rand")
#> [1] FALSE
RNGkind(normal.kind = "user")
#> Error in RNGkind(normal.kind = "user"): 'user_norm_rand' not in load table
```

When I look at `RcppZiggurat.so` using `nm`, `objectdump` or `readelf`, it lists `user_norm_rand` among the available symbols. 

The example from `help(Random.user)` works without problems.

**Comment by eddelbuettel** [comment]
It's been a while since I tried this.  I think it used to work, and I am wondering if the function registration now gets in the way...

**Comment by eddelbuettel** [comment]
And indeed it still works with version 0.1.2 (random pick I made).  

And I see `user_norm_rand` in the shared library when I call `nm` on it.  R calls this directly, so I am not sure how we get in the way, but we do. 

Let me ponder this.

**Comment by eddelbuettel** [comment]
It is missing in the registration setup.  I think I can fix that -- good catch!

**Comment by rstub** [comment]
I have only now been able to test your patch https://github.com/eddelbuettel/rcppziggurat/commit/edfbfc230bc567f448a7c943e7544229cb58acd3. The R integration is working fine, which is why I am closing this issue. Unfortunately the speed increase is not as impressive as with `zrnormR` or `zrnorm`:

``` r
library(RcppZiggurat)
is.loaded("user_norm_rand")
#> [1] TRUE
system.time(rnorm(10^8))
#>        User      System verstrichen 
#>       6.040       0.124       6.165
system.time(zrnormR(10^8))
#>        User      System verstrichen 
#>       1.104       0.120       1.223
system.time(zrnorm(10^8))
#>        User      System verstrichen 
#>       0.656       0.128       0.782
RNGkind(normal.kind = "user")
system.time(rnorm(10^8))
#>        User      System verstrichen 
#>       3.132       0.132       3.263
```

**Comment by eddelbuettel** [comment]
Thank you for that too!  I had been meaning to circle back to you.

I think I did the same / similar benchmark test when I first cooked this up and was similarly disappointed.  Thinking about a little I realized that ... this is likely due to the overhead from R's state keeping which is much more involved than what we do here with Ziggurat.    Still a gain though...

I re-opened just to remind myself to release this.  I need one more pass over the vignette, not sure when I get to that.   I could drop the package into a drat repo if that helped you, but you seem fine with GitHub installation too :)

Gruesse aus Chicago.
